CMake depreciation

Hi,
When trying to install catkit2 on CMake 3.27.6 with the guide at the end of [README.md](https://github.com/spacetelescope/catkit2), when executing setup.py, this error occurs:

```
CMake Error at CMakeLists.txt:3 (project):
  Generator

    NMake Makefiles

  does not support platform specification, but platform

    x64

  was specified.


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
Traceback (most recent call last):
  File "setup.py", line 145, in <module>
    install_requires=[],
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\__init__.py",
line 87, in setup
    return distutils.core.setup(**attrs)
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\_distutils\core.py",
line 185, in setup
    return run_commands(dist)
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\_distutils\core.py",
line 201, in run_commands
    dist.run_commands()
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\_distutils\dist.py",
line 969, in run_commands
    self.run_command(cmd)
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\dist.py",
line 1208, in run_command
    super().run_command(command)
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\_distutils\dist.py",
line 988, in run_command
    cmd_obj.run()
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\command\develop.py",
line 34, in run
    self.install_for_development()
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\command\develop.py",
line 114, in install_for_development
    self.run_command('build_ext')
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\_distutils\cmd.py",
line 318, in run_command
    self.distribution.run_command(command)
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\dist.py",
line 1208, in run_command
    super().run_command(command)
  File
"C:\Users\LEO\anaconda3\envs\catkit2\lib\site-packages\setuptools\_distutils\dist.py",
line 988, in run_command
    cmd_obj.run()
  File "setup.py", line 92, in run
    self.build_extension(ext)
  File "setup.py", line 118, in build_extension
    subprocess.check_call(['cmake', ext.sourcedir] + cmake_args,
cwd=self.build_temp, env=env)
  File "C:\Users\LEO\anaconda3\envs\catkit2\lib\subprocess.py", line 363,
in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake',
'C:\\Users\\LEO\\Documents\\OSAE\\Project_TDH2\\Repository\\catkit2',
'-DPYTHON_EXECUTABLE=C:\\Users\\LEO\\anaconda3\\envs\\catkit2\\python.exe',
'-A', 'x64']' returned non-zero exit status 1.
```

--- a/docs/installation.rst
+++ b/docs/installation.rst
@@ -28,7 +28,7 @@ At this point, all C++ and Python dependencies of catkit2 should have been downl
 
     python setup.py develop
 
-This step can fail if environment variable for cmake are not well-defined. Defining `CMAKE_GENERATOR` with a value of a generator in the available generators (which can be found with `cmake --help`) should fix the error. If the error persists, try to reboot to apply the environment variable (needed on Windows).
+This will use the default CMake generator to compile catkit_core and its Python bindings. If the default generator doesn't support 64bit compilation, this step will return an error and you will need to specify a default generator to use by setting the CMAKE_GENERATOR environment variable to your preferred generator. You can list all generators installed on your machine with cmake --help. You will have to restart your terminal after changing your environment variables as usual.
 
 You should see the main compilation complete in the terminal output without errors. We installed catkit2 in editable mode to make it easier to apply updates. Any updates to the core library requires recompilation of the bindings, which can be done by simply reinstalling the package:
 

